### PR TITLE
[ISSUE #2708] fix offset back to minLogicOffset when timeout or connect fail

### PR DESCRIPTION
**Pull request by EZLippi**
## What is the purpose of the change

issue #2708 

## Brief changelog

if connect or read timeout occur when fetch offset from broker, queue offset will be reset to minLogicOffset as issue(#2708) showed , the way to fix this is to take care of RemotingException specially, if queryOffsetRequest got timeout, we should execute pullRequestLater other then use -1 to request. 

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- 

- [x]  Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 

- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPushConsumerImpl.java [hunk 1]
```diff
@@ -271,6 +271,10 @@ public void pullMessage(final PullRequest pullRequest) {
             if (processQueue.isLocked()) {
                 if (!pullRequest.isLockedFirst()) {
                     final long offset = this.rebalanceImpl.computePullFromWhere(pullRequest.getMessageQueue());
+                    if (offset == -3) {
+                      this.executePullRequestLater(pullRequest, pullTimeDelayMillsWhenException);
+                      log.info("pull message later because remoting exception with broker, {}", pullRequest);
+                    }
```
Comment (caigy) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultMQPushConsumerImpl.java:277]: Should the procedure just return here?

**Review by caigy — CHANGES_REQUESTED** [review]
Please resolve conflicts first.

**Comment by areyouok** [comment]
I think this issue is resolved in 2 PRs:

https://github.com/apache/rocketmq/pull/2948

https://github.com/apache/rocketmq/pull/3158

So I close this PR.

**Comment by LvChenhui** [comment]
> I think this issue is resolved in 2 PRs:
> 
> #2948
> 
> #3158
> 
> So I close this PR.

你好，
我也查看了你说的这两个pr，但是我和楼主一样发现了当RemoteBrokerOffsetStore#readOffset时就出现了exception，会被默认设置为-2，外层offset则会等于-1
image

**Comment by zhengtenghui** [comment]
同楼上 @LvChenhui

**Comment by zhengtenghui** [comment]
但是楼主的代码是不是少了个return @EZLippi

**Pull request closed without merging**